packages/core/src/routes/admin-user.ts

```diff
@@ -0,0 +1,53 @@
+import { userInfoSelectFields } from '@logto/schemas';
+import pick from 'lodash.pick';
+import { InvalidInputError } from 'slonik';
+import { object, string } from 'zod';
+
+import koaGuard from '@/middleware/koa-guard';
+import { findRolesByRoleNames } from '@/queries/roles';
+import { findAllUsers, findUserById, updateUserById } from '@/queries/user';
+
+import { AuthedRouter } from './types';
+
+export default function adminUserRoutes<T extends AuthedRouter>(router: T) {
+  router.get('/users', async (ctx, next) => {
+    const users = await findAllUsers();
+    ctx.body = users.map((user) => pick(user, ...userInfoSelectFields));
+
+    return next();
+  });
+
+  router.patch(
+    '/users/:userId/roleNames',
+    koaGuard({
+      params: object({ userId: string().min(1) }),
+      body: object({ names: string().array() }),
+    }),
+    async (ctx, next) => {
+      const {
+        params: { userId },
+        body: { names },
```

Comment:
better keep body key name equals to path name, how about:
`PATCH /users/:userId/roleNames`
and body:
```json
{
  "roleNames": ["admin"]
}
```
I personally prefer as less schemas as possible